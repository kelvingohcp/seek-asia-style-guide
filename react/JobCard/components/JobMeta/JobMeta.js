import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
import SellingPoint from '../SellingPoint/SellingPoint';
import ShelfLink from '../ShelfLink/ShelfLink';
import Description from '../Description/Description';
import { JobCardPropTypes } from '../../JobCardPropTypes';

const JobMeta = ({
  sellingPoints,
  isSplitView,
  showSellingPoint,
  description,
  showDescription,
  job,
  shelfSectionOpen,
  onClick
}) => {
  return (
    <Fragment>
      <SellingPoint
        sellingPoints={sellingPoints}
        showSellingPoint={showSellingPoint}
      />
      {!isSplitView && (
        <Description
          description={description}
          showDescription={showDescription}
        />
      )}
      <ShelfLink
        job={job}
        shelfSectionOpen={shelfSectionOpen}
        mobileOnly={!isSplitView}
        onClick={onClick}
      />
    </Fragment>
  );
};

JobMeta.propTypes = {
  sellingPoints: PropTypes.arrayOf(PropTypes.string),
  isSplitView: PropTypes.bool,
  showSellingPoint: PropTypes.bool,
  bannerUrl: PropTypes.string,
  description: PropTypes.string,
  showDescription: PropTypes.bool,
  job: JobCardPropTypes.job,
  shelfSectionOpen: PropTypes.bool,
  onClick: PropTypes.func,
  isMobile: PropTypes.bool
};

export default JobMeta;
