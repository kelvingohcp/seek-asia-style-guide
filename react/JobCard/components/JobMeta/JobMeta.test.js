import React from 'react';
import { shallow } from 'enzyme';
import JobMeta from './JobMeta';

jest.mock('../../JobCardPropTypes', () => {
  return {
    JobCardPropTypes: {
      job: {}
    }
  };
});

const sellingPoints = [
  'sellingPoints 1',
  'sellingPoints 2',
  'sellingPoints 3'
];
const description = 'dummyDescription';

describe('JobMeta', () => {
  it('should render correctly', () => {
    expect(shallow(<JobMeta sellingPoints={sellingPoints} isSplitView={true} showSellingPoint={true} enableBrandedAd={true} description={description} showDescription={true} shelfSectionOpen={false} />)).toMatchSnapshot();
  });
});
