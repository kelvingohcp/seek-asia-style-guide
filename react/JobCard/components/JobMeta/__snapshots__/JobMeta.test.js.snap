// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JobMeta should render correctly 1`] = `
<Fragment>
  <SellingPoint
    enableBrandedAd={true}
    isSplitView={true}
    sellingPoints={
      Array [
        "sellingPoints 1",
        "sellingPoints 2",
        "sellingPoints 3",
      ]
    }
    showSellingPoint={true}
  />
  <ShelfLink
    mobileOnly={false}
    shelfSectionOpen={false}
  />
  <JobLabel />
</Fragment>
`;
